packages/transport/src/chrome-runtime/client-connection-manager.ts

```
@@ -0,0 +1,183 @@
+     * request.  Messages are forwarded to the page-provided client channel
+     * without much validation.
+     *
+     * The extension may provide a new channel name, to be used as a source for a
+     * stream that gets transferred to the client.
+     *
+     * TODO: response origin?
+     *
+     * @param ev any message transmitted on the client-specific service channel
+     * by any part of the extension runtime, hopefully just the service router
+     */
+    const serviceListener: Parameters<chrome.runtime.PortMessageEvent['addListener']>[0] = (
+      ev: unknown,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      _: chrome.runtime.Port,
+    ) => {
+      if (!isTransportData(ev)) throw Error('Unknown data in transport');
+      const response = ev;
+      if (isTransportMessage(response)) {
+        const { requestId, message } = response;
+        clientPort.postMessage({ requestId, message } satisfies TransportMessage);
+      } else if (isTransportInitChannel(response)) {
+        const { requestId, channel: name } = response;
+        const sourcePort = chrome.runtime.connect({ includeTlsChannelId: true, name });
+        const stream = new ReadableStream(new ChromeRuntimeStreamSource(sourcePort));
+        clientPort.postMessage({ requestId, stream } satisfies TransportStream, [stream]);
+      } else throw Error('Unimplemented response kind');
+    };
+
+    this.connections.set(
+      clientId,
+      Object.assign(connection, {
+        clientPort,
+        servicePort,
+        clientListener,
+        serviceListener,
+      } satisfies ClientConnection),
+    );
+
+    servicePort.onMessage.addListener(serviceListener);
+    servicePort.onDisconnect.addListener(() => this.endConnection(clientId));
+
+    clientPort.addEventListener('message', clientListener);
+    clientPort.addEventListener('messageerror', () => this.endConnection(clientId));
```

Does this mean the errors get swallowed silently? Are we losing anything here that we should be reporting back to the requestor?